### PR TITLE
Add ncvue

### PR DESCRIPTION
**Pull request by mcuntz**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mcuntz** [comment]
Dear @conda-forge/help-python 

I try to add a GUI application, which uses tkinter. macOS and win64 tests work fine.

I had to add tk to the run requirements on linux and xorg-libx11 as test requirements on linux as well. However, it is still failing with the import error:
ImportError: Cannot load backend 'TkAgg' which requires the 'tk' interactive framework, as 'headless' is currently running

Can you please tell me how to get out of this headless state of mind ;-)

Kind regards,
Matthias

**Comment by mcuntz** [comment]
Dear @conda-forge/help-python 

I further tried to get it working but it does not work on the linux staged-recipes (only). As said, ncvue is a GUI application (a plotting tool) and is opening a tkinter window. So on linux staged-recipes it tells me that it is in a "headless" state. Tests work on the osx staged-recipes and the win_64  staged-recipes.

I tried to add 
`  script_env:
   - MPLBACKEND=TkAgg`
in the build section.

I also tried to add an `app` section, adding an icon file and a launch script to the recipes folder.
But all to no avail :-(

Could you have a look over the meta.yaml file, please?

Kind regards,
Matthias

**Comment by mcuntz** [comment]
Just read that my ping might not work yet  @conda-forge/help-python

@conda-forge-admin, please ping team conda-forge/help-python

**Comment by mcuntz** [comment]
@conda-forge-admin, please ping team

**Comment by saraedum** [comment]
@mcuntz does it work for you (on Linux) if you build the recipe locally? I mean, is this only a CI issue during the testing somehow or is it actually broken on Linux?

File: recipes/ncvue/ncvue [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python
```
Comment (carterbox) [inline recipes/ncvue/ncvue:1]: This file should go in your project as `src/ncvue/__main__.py`. This allows anyone to launch your app by calling `python -m nvcue`.

File: recipes/ncvue/ncvue [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    if len(ncfile) > 0:
+        ncfile = ncfile[0]
+    else:
+        ncfile = ''
+
+    # This must be before any other call to matplotlib
+    # because it uses the TkAgg backend.
+    # This means, do not use --pylab with ipython.
+    ncvue(ncfile=ncfile, miss=miss)
```
Comment (carterbox) [inline recipes/ncvue/ncvue:57]: After moving the file into your python module as __main__.py, move this code into a main() function and then register this main() function as a console_script entry point.
Then pip will automatically wrap this entry point with a bash/bat script that will make the function available on the command line without calling the python interpreter directly.

https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html

File: recipes/ncvue/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    - tk  # [linux]
+
+test:
+  imports:
+    - ncvue
+  commands:
+    - pip check
+  requires:
+    - pip
+    - xorg-libx11  # [linux]
```
Comment (carterbox) [inline recipes/ncvue/meta.yaml:47]:
```suggestion
```
You need to include a `yum_requirements.txt` file if you want to load any libraries that link against opengl.

https://conda-forge.org/docs/maintainer/knowledge_base.html#libgl

**Review by carterbox — CHANGES_REQUESTED** [review]
The icon file is already available in the source downloaded from pypi. Use the path to that file instead of including a copy of it here.

**Comment by mcuntz** [comment]
> @mcuntz does it work for you (on Linux) if you build the recipe locally? I mean, is this only a CI issue during the testing somehow or is it actually broken on Linux?

Dear @saraedum , it actually works on Linux if build locally. It is only a CI issue during the testing.

File: recipes/ncvue/ncvue (hunk 2 above)
Comment (mcuntz) [reply recipes/ncvue/ncvue:57]: @carterbox Created a gui_scripts entry_point. Thanks for the hint.

**Comment by jakirkham** [comment]
@conda-forge-admin, please ping team

File: recipes/ncvue/meta.yaml (hunk 3 above)
Comment (mcuntz) [reply recipes/ncvue/meta.yaml]: Hey @carterbox

I added a yum_requirements.txt file with the entries:
```
mesa-libGL
mesa-dri-drivers
libselinux
libXdamage
libXxf86vm
libXext
xorg-x11-server-Xorg
```
But the build still fails with the messsage:
`ImportError: Cannot load backend 'TkAgg' which requires the 'tk' interactive framework, as 'headless' is currently running`
???

**Comment by mcuntz** [comment]
@conda-forge-admin please rerender

File: recipes/ncvue/meta.yaml (hunk 3 above)
Comment (carterbox) [reply recipes/ncvue/meta.yaml]: Sorry, but I'm not familiar with the various backends for matplotlib or their requirements. What happened when you switched to Agg the logs are missing?

Are you trying to open GUI elements on our headless servers?

**Comment by mcuntz** [comment]
@carterbox
I basically removed the `python -c 'import ncvue'` from Linux but left it on Win and Mac, if this is acceptable. I will hence also remove the yum_requirements.txt file again.

Could you also give me a hint how to put the correct path to the icon file? You said not to put it again here but use it from the tar.gz. I tried:
`icon: ncvue_icon.png`
`icon: src/ncvue/images/ncvue_icon.png`
`icon: app/src/ncvue/images/ncvue_icon.png`
but nothing worked. The file ncvue_icon.png is, obviously, in the subfolder `src/ncvue/images`  under the main folder.

**Review by carterbox — APPROVED** [review]
I was wrong, the search for icon does not look in the unpackaged source. This is unexpected, and should probably be fixed.

It's fine to skip tests on linux because your app is unable to be run in headless mode on linux, but this is something your will have to keep an eye on for future releases.

File: recipes/ncvue/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    - pyshp
+    - python
+    - scipy
+    - shapely
+    - six
+    - tk  # [linux]
+
+test:
+  commands:
+    - export MPLBACKEND=Agg  # [linux]
```
[the remark this answers is not in the document]
Comment (mcuntz) [reply recipes/ncvue/meta.yaml]: @carterbox 
Thanks for all the feedbacks and the merge.

**Pull request merged**